packages/backend/src/singletons/Synchronizer.mjs

```diff
@@ -246,6 +250,21 @@ export class Synchronizer extends EventEmitter {
         throw err
       }
     }
+
+    for (const blockNum of blockNumbers) {
```

Comment:
This looks correct, but it needs to happen before the loop on line 222. You want to load the block timestamp before calling the event handler so that it's available for the handler.

Right now if you have, for example, you're calling `handleStateTreeLeaf` and then creating the `BlockTimestamp` after. So `handleStateTreeLeaf` can't access the `BlockTimestamp`.

Reply:
🎉 I have timestamps! 🎉

<img width="1231" alt="Screen Shot 2023-01-30 at 12 14 40 PM" src="https://user-images.githubusercontent.com/79813968/215586389-580c2913-6017-4a9c-998a-28bb02b81a2c.png">

<img width="881" alt="Screen Shot 2023-01-30 at 1 14 47 PM" src="https://user-images.githubusercontent.com/79813968/215586304-9916c7d5-4b2b-43c1-be41-517d888ff0d0.png">

Reply:
@vimwitch sorry just tagging you cause not sure if you saw after I edited

Reply:
Sure, I think it's fine to merge as main is a bit wip. I'll do a proper review here in a moment